native/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java

```diff
@@ -0,0 +1,629 @@
+        initStream.addRequiredParameter(
+                TypeDescriptor.getStreamTypeDescriptorNode(
+                        SyntaxTreeConstants.SYNTAX_TREE_VAR_ANYDATA,
+                        TypeDescriptor.getOptionalTypeDescriptorNode(BalFileConstants.EMPTY_STRING,
+                                BalFileConstants.ERROR)),
+                BalFileConstants.ANYDATA_STREAM);
+        clientStream.addMember(initStream.getFunctionDefinitionNode(), true);
+
+        Function nextStream = new Function(BalFileConstants.NEXT);
+        nextStream.addQualifiers(new String[]{BalFileConstants.KEYWORD_PUBLIC, BalFileConstants.KEYWORD_ISOLATED});
+        nextStream.addReturns(NodeParser.parseTypeDescriptor(String.format(
+                BalFileConstants.NEXT_STREAM_RETURN_TYPE,
+                String.format(BalFileConstants.RECORD_FIELD_VAR, BalFileConstants.ENTITIES, entity.getEntityName()))));
+        nextStream.addStatement(NodeParser.parseStatement(BalFileConstants.NEXT_STREAM_STREAM_VALUE));
+
+        IfElse streamValueNilCheck = new IfElse(NodeParser.parseExpression(
+                BalFileConstants.NEXT_STREAM_IF_STATEMENT));
+        streamValueNilCheck.addIfStatement(NodeParser.parseStatement(
+                BalFileConstants.NEXT_STREAM_RETURN_STREAM_VALUE));
+        IfElse streamValueErrorCheck = new IfElse(NodeParser.parseExpression(
+                BalFileConstants.NEXT_STREAM_ELSE_IF_STATEMENT));
+        streamValueErrorCheck.addIfStatement(NodeParser.parseStatement(
+                BalFileConstants.NEXT_STREAM_RETURN_STREAM_VALUE));
+        streamValueErrorCheck.addElseStatement(NodeParser.parseStatement(String.format(
+                BalFileConstants.NEXT_STREAM_ELSE_STATEMENT,
+                String.format(BalFileConstants.RECORD_FIELD_VAR, BalFileConstants.ENTITIES, entity.getEntityName()),
+                String.format(BalFileConstants.RECORD_FIELD_VAR, BalFileConstants.ENTITIES, entity.getEntityName()))));
+        streamValueErrorCheck.addElseStatement(NodeParser.parseStatement(BalFileConstants.RETURN_NEXT_RECORD));
+        streamValueNilCheck.addElseBody(streamValueErrorCheck);
+        nextStream.addIfElseStatement(streamValueNilCheck.getIfElseStatementNode());
+        clientStream.addMember(nextStream.getFunctionDefinitionNode(), true);
+
+        Function closeStream = new Function(BalFileConstants.CLOSE);
+        closeStream.addQualifiers(new String[]{BalFileConstants.KEYWORD_PUBLIC, BalFileConstants.KEYWORD_ISOLATED});
+        closeStream.addReturns(TypeDescriptor.getOptionalTypeDescriptorNode(BalFileConstants.EMPTY_STRING,
+                BalFileConstants.ERROR));
+        closeStream.addStatement(NodeParser.parseStatement(BalFileConstants.CLOSE_STREAM_STATEMENT));
+        clientStream.addMember(closeStream.getFunctionDefinitionNode(), true);
+
+        return clientStream;
+    }
+
+    public static SyntaxTree generateConfigBalFile() {
```

Comment:
Shall we rename the function to `generateConfigSyntaxTree`?

Reply:
Updated in [d6a53d3](https://github.com/ballerina-platform/persist-tools/pull/15/commits/d6a53d3f948c8f076acb3cc1f2730258a998cf40)